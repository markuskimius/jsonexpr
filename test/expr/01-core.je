myvar = "Hello, world!";
myarr = [ 101, 211, 321, THIS[2]*2 ];
myobj = {
    "name"  : "Bob",
    "email" : THIS.name + "@" + THIS.name,
    "age"   : 30,
};

PRINT(null);
PRINT(true);
PRINT(false);
PRINT(0);
PRINT(1);
PRINT(-1);
PRINT(03777777777777777777777);
PRINT(0xffffffffffffffff);
PRINT(0b11111111111111111111111111111111111111111111111111111111111111111111111111111111);

PRINT(myvar);
PRINT(myarr);
PRINT(myobj);
PRINT(myarr[1]);
PRINT(myobj["email"]);
PRINT(myobj.age);

myarr[5] = 431;
myarr[7] = myarr[5] + 7000;
myobj.age = myobj["age"] + 1;
myobj.phone = "111-111-1111";

PRINT(myarr);
PRINT(myobj);
