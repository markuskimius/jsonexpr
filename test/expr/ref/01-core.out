null
true
false
0
1
-1
-1
-1
-1
Hello, world!
[ 101, 211, 321, 642 ]
{ "age": 30, "email": "Bob@Bob", "name": "Bob" }
211
Bob@Bob
30
[ 101, 211, 321, 642, null, 431, null, 7431 ]
{ "age": 31, "email": "Bob@Bob", "name": "Bob", "phone": "111-111-1111" }
